snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs

```diff
@@ -1,49 +1,42 @@
 ﻿// <Snippet8>
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 class Program
 {
     static void Main()
-   {
-      object[] items = { new Book("The Tempest"), new Person("John") };
-      foreach (var item in items) {
-        if (item is var obj)
-          Console.WriteLine($"Type: {obj.GetType().Name}, Value: {obj}"); 
-      }
-   }
-}
-
-class Book
-{
-    public Book(string title) 
     {
-       Title = title;    
-    }
+        int[] testSet = { 100271, 234335, 342439, 999683 };
 
-    public string Title { get; set; }
+        var primes = testSet.Where(n => Factor(n).ToList() is var factors
+                                    && factors.Count == 2
+                                    && factors.Contains(1)
+                                    && factors.Contains(n));
```

Comment:
The last two checks are not necessary: `1` and `n` are always factors of `n`. So, it looks those two checks are only for the sake of the example of the need of the `factors` variable?

Reply:
Good point! But as you say removing them means we don't need the `factors` variable which is the point of the example. 
I could change the example to do something else with the factors, say find numbers with between 3 and 5 factors, if you think this example is a bit misleading?

Reply:
I agree that the extra checks are unnecessary, but they do clarify the code.

I'm ready to approve this, pending the discussion on dotnet/docs#17197